pkg/queue/priority_queue.go

```diff
@@ -0,0 +1,156 @@
+package queue
+
+import "container/heap"
+
+// PriorityQueue queue has priority, the first element has the highest priority
+type PriorityQueue struct {
```

Comment:
Why do we need to implement it by ourselves? Can we use the existed library?

Reply:
it is based on https://github.com/facebookarchive/pqueue/blob/master/pqueue.go